rts/rts.h

```diff
@@ -178,6 +178,13 @@ void $POP();
 
 extern $Msg timerQ;
 
+struct ThreadStore {
```

Comment:
Good idea!

One could also consider using multiple specific keys (beyond self_key that is) for storing additional fields. That could possibly be a tad more efficient, and it would also remove the potential hazards of storing pointers to ThreadStore variables that might go out of scope. (I don't think there's any problem at the moment, though!)

Reply:
That's a cool idea. Didn't know one could do that. I'm reverting parts of my PR and will focus on doing this. No need for multiple keys in this patch but for the future...